### PR TITLE
HWKAPM-532 | Display service name in Distributed Tracing graph

### PR DESCRIPTION
**Pull request by pavolloffay**
https://issues.jboss.org/browse/HWKAPM-532

### CODE REVIEW

**Comment by pavolloffay** [comment]
@objectiser  could you please review?

**Comment by pavolloffay** [comment]
retest this please

File: api/src/main/java/org/hawkular/apm/api/model/analytics/CommunicationSummaryStatistics.java [hunk 1]
```diff
@@ -196,9 +202,14 @@ public void setOutbound(Map<String, ConnectionStatistics> outbound) {
...
+    public String getServiceName() {
+        return serviceName;
+    }
+
+    public void setServiceName(String serviceName) {
+        this.serviceName = serviceName;
+    }
+
     @Override
     public String toString() {
```
Comment (objectiser) [inline api/src/main/java/org/hawkular/apm/api/model/analytics/CommunicationSummaryStatistics.java]: Minor thing - can you add service name to toString,hashCode and equals

File: server/elasticsearch/src/test/java/org/hawkular/apm/server/elasticsearch/AnalyticsServiceElasticsearchTest.java [hunk 2]
```diff
@@ -2871,6 +2875,64 @@ public void testGetCommunicationSummaryStatisticForHost() {
...
+                .setBusinessTransaction("testapp");
+
+        Collection<CommunicationSummaryStatistics> communicationSummaryStatisticsList =
+                analytics.getCommunicationSummaryStatistics(null, criteria, false);
+
+        Assert.assertEquals(1, communicationSummaryStatisticsList.size());
+        CommunicationSummaryStatistics communicationSummaryStatistics =
+                communicationSummaryStatisticsList.iterator().next();
+
+        Assert.assertNotNull("wildfly", communicationSummaryStatistics.getServiceName());
```
Comment (objectiser) [inline server/elasticsearch/src/test/java/org/hawkular/apm/server/elasticsearch/AnalyticsServiceElasticsearchTest.java]: assertEquals?

File: server/elasticsearch/src/test/java/org/hawkular/apm/server/elasticsearch/AnalyticsServiceElasticsearchTest.java [hunk 3]
```diff
@@ -2871,6 +2875,64 @@ public void testGetCommunicationSummaryStatisticForHost() {
...
```
Comment (objectiser) [inline server/elasticsearch/src/test/java/org/hawkular/apm/server/elasticsearch/AnalyticsServiceElasticsearchTest.java]: assertEquals?

File: ui/src/main/scripts/plugins/e2e/ts/e2e.ts [hunk 4]
```diff
@@ -77,7 +77,7 @@ module E2E {
       $scope.operations = [];
 
       _.each($scope.e2eData, (node) => {
-        $scope.topLevel.push(node.id);
+        $scope.topLevel.push(node.serviceName + ': ' + node.id);
```
Comment (objectiser) [inline ui/src/main/scripts/plugins/e2e/ts/e2e.ts]: what if service isn't set?

**Comment by objectiser** [comment]
@pavolloffay LGTM - some minor comments, but main thing is the UI needs to cope with no service name.

File: ui/src/main/scripts/plugins/e2e/ts/e2e.ts (hunk 4 above)
Comment (pavolloffay) [inline ui/src/main/scripts/plugins/e2e/ts/e2e.ts]: good point https://github.com/hawkular/hawkular-apm/pull/585/files needs to go first, then I will change this to dealt with no service name.

**Comment by pavolloffay** [comment]
PR updated and rebased @objectiser could you please review?

**Comment by objectiser** [comment]
@pavolloffay Works great - ok to merge?

**Comment by pavolloffay** [comment]
yes please :+1:, thanks :)

**Pull request merged**